pages/create-entry.js

```diff
@@ -14,103 +14,83 @@ import {
 
-    function checkListItemInput() {
-      let value = event.target.value;
+  //HANDLE FUNCTIONS
+  function handleItemInput(event) {
+    const value = event.target.value;
+    const inListItems = itemInListItems(value.trim());
 
-      if (!value.startsWith(" ") && value.length > 0) {
-        setListItemExistsInListItems(listItemInListItems(value));
-        setSubmitButtonReady(!listItemInListItems(value));
-      } else if (value.length > 0) {
-        value = value.trim();
-        event.target.value = value;
-        checkListItemInput();
+    if (testHasChar.test(value)) {
+      if (!inListItems) {
+        setItemNameValidated(true);
+        if (currentInfo[0] === "listItem") setCurrentInfo(["", ""]);
+        setSubmitButtonReady(categoryDropdownUsed || categoryValidated);
       } else {
+        setCurrentInfo(["listItem", "Dieser Eintrag existiert bereits"]);
+        setItemNameValidated(false);
         setSubmitButtonReady(false);
-        setListItemExistsInListItems(false);
       }
+    } else {
+      setItemNameValidated(false);
+      setSubmitButtonReady(false);
+      setCurrentInfo([
+        "listItem",
+        "Ein Eintrag muss mind. ein Zeichen enthalten",
+      ]);
     }
   }
 
   function handleCategoryInput(event) {
-    checkCategoryInput();
+    const value = event.target.value;
+    const inCategories = categoryInCategories(value.trim());
 
-    function checkCategoryInput() {
-      if (!event.target.value.startsWith(" ")) {
-        setCategoriesSelectionAvailable(false);
+    if (testHasChar.test(value)) {
+      setCategoryDropdownUsed(false);
 
-        if (categoryInCategories(event.target.value)) {
-          setCategoryExistsInCategories(true);
-        } else {
-          setCategoryExistsInCategories(false);
-        }
+      if (!inCategories) {
+        setCategoryValidated(true);
+        if (currentInfo[0] === "category") setCurrentInfo(["", ""]);
+        setSubmitButtonReady(itemNameValidated);
       } else {
-        const value = event.target.value.substring(
-          1,
-          event.target.value.length
-        );
-        event.target.value = value;
-        checkCategoryInput();
-        setCategoryExistsInCategories(false);
-        setCategoriesSelectionAvailable(true);
+        setCurrentInfo(["category", "Diese Kategorie existiert bereits"]);
+        setCategoryValidated(false);
+        setSubmitButtonReady(false);
+      }
+    } else {
+      setCategoryValidated(false);
+
+      if (value.length > 0) {
+        setCurrentInfo([
+          "category",
+          "Eine Kategorie muss mind. ein Zeichen enthalten",
+        ]);
+        setCategoryDropdownUsed(false);
+        setSubmitButtonReady(false);
+      } else {
+        if (currentInfo[0] === "category") setCurrentInfo(["", ""]);
+        setCategoryDropdownUsed(true);
+        setSubmitButtonReady(itemNameValidated && categoryDropdownUsed);
       }
     }
   }
```

Comment:
I am sure you can simplify this as well. As an outside dev I would need a long time to parse this control flow. Try to keep it simple!